libs/partners/openai/langchain_openai/embeddings/base.py

```diff
@@ -586,3 +543,73 @@ async def aembed_query(self, text: str) -> List[float]:
         """
         embeddings = await self.aembed_documents([text])
         return embeddings[0]
+
+    def _get_embeddings(
+        self,
+        input: Sequence[Union[List[int], str]],
+        *,
+        chunk_size: Optional[int] = None,
+    ) -> List[List[float]]:
+        embeddings: List[List[float]] = []
+        chunk_size = chunk_size or self.chunk_size
+        _iter: Iterable = range(0, len(input), chunk_size)
+        if self.show_progress_bar:
+            try:
+                from tqdm.auto import tqdm
+            except ImportError:
+                pass
+            else:
+                _iter = tqdm(_iter)
+        for chunk_start in _iter:
+            chunk_end = chunk_start + chunk_size
+            response = self.client.create(
+                input=input[chunk_start:chunk_end], **self._invocation_params
+            )
+            if not isinstance(response, dict):
+                if hasattr(response, "model_dump"):
+                    response = response.model_dump()
+                else:
+                    response = response.dict()
+            embeddings.extend(r["embedding"] for r in response["data"])
+        return embeddings
+
+    async def _aget_embeddings(
+        self,
+        input: Sequence[Union[List[int], str]],
+        *,
+        chunk_size: Optional[int] = None,
+    ) -> List[List[float]]:
+        chunk_size = chunk_size or self.chunk_size
+        responses = await asyncio.gather(
+            *(
+                self.async_client.create(
+                    input=input[start : start + chunk_size], **self._invocation_params
+                )
+                for start in range(0, len(input), chunk_size)
+            )
+        )
```

Comment:
@nfcampos does this look right

Reply:
This does all requests in parallel without a limit, not sure if that's intended?

Reply:
++ do we want to set a limit on parallelism? I think this could end up rate limiting pretty quick

Reply:
@nfcampos is the right way to do this using an `asyncio.Semaphore(paralellism)` like this?

Would be great to land this

```suggestion
        semaphore = asyncio.Semaphore(parallelism)
        async def create_limit_parallelism(inputs: List) -> List:
            async with semaphore:
                return await self.async_client.create(input=inputs, **self._invocation_params)
        responses = await asyncio.gather(
            *(
                create_limit_parallelism(input[start : start + chunk_size])
                for start in range(0, len(input), chunk_size)
            )
        )
```